pycyphal/transport/can/media/socketcand/_socketcand.py

```diff
@@ -0,0 +1,356 @@
+    of the other PythonCAN compatible interfaces take. In the future, hopefully this Media type
+    will be implimented without the use of PythonCAN at all.
+
+"""
+
+from __future__ import annotations
+import queue
+import time
+import typing
+import asyncio
+import logging
+import threading
+from functools import partial
+import dataclasses
+import warnings
+
+import can
+from pycyphal.transport import Timestamp, ResourceClosedError, InvalidMediaConfigurationError
+from pycyphal.transport.can.media import Media, FilterConfiguration, Envelope, FrameFormat, DataFrame
+
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass(frozen=True)
+class _TxItem:
+    msg: can.Message
+    timeout: float
+    future: asyncio.Future[None]
+    loop: asyncio.AbstractEventLoop
+
+
+@dataclasses.dataclass(frozen=True)
+class PythonCANBusOptions:
+    hardware_loopback: bool = False
+    """
+    Hardware loopback support.
+        If True, loopback is handled by the supported hardware.
+        If False, loopback is emulated with software.
+    """
+    hardware_timestamp: bool = False
+    """
+    Hardware timestamp support.
+        If True, timestamp returned by the hardware is used.
+        If False, approximate timestamp is captured by software.
+    """
+
+
+class SocketcandMedia(Media):
+    # pylint: disable=line-too-long
+    """
+    Media interface adapter for `Socketcand <https://github.com/linux-can/socketcand/tree/master>` using the
+    built in interface compatibility from `Python-CAN <https://python-can.readthedocs.io/>`.
+    Please refer to the Socketcand documentation and the Python-CAN documentation for information about
+    supported hardware, configuration, and installation instructions.
+
+    This media interface supports both Classic CAN and CAN FD. The selection logic is documented below.
+
+    Here is a basic usage example based on the Yakut CLI tool.
+    Suppose you have two computers:
+    One connected to a CAN capable device and that computer is able to connect and recieve CAN data from the
+    CAN device. Using socketcand with a command such as `socketcand -v -i can0 -l 123.123.1.123`
+    on this first computer will bind it too a socket (default port for socketcand is 29536, so it is also default here).
+
+    On your second computer:
+
+        export UAVCAN__CAN__IFACE="socketcand:can0:123.123.1.123"
+        export UAVCAN__CAN__BITRATE='500000'
+        export UAVCAN__CAN__MTU=8
+        yakut sub 33:uavcan.si.unit.voltage.scalar
+
+    This will allow you to wirelessly recieve CAN data on computer two through the wired connection on computer 1.
+
+    """
+
+    _MAXIMAL_TIMEOUT_SEC = 0.1
+
+    def __init__(
+        self,
+        iface_name: str,
+        bitrate: typing.Union[int, typing.Tuple[int, int]],
+        mtu: typing.Optional[int] = None,
+        *,
+        loop: typing.Optional[asyncio.AbstractEventLoop] = None,
```

Comment:
Please remove

Reply:
Is the reason you are suggesting removing these parameters because of the documentation here that only shows interface, host, port, and channel: https://python-can.readthedocs.io/en/stable/interfaces/socketcand.html ?. The reason I ask is because I believe socketcand can still take either FD or normal frames, but I may be mistaken.

Reply:
Also, the media class contains an abstract method for mtu, I could implement it for the sake of functionality and return some default value, but it seems like mtu should remain

Reply:
I just looked at the python-can socketcanDaemonBus instantiation, I guess it does not require these params, my mistake

Reply:
Sorry, I was unclear. I am not sure if `bitrate` and `mtu` are required --- if they aren't, they obviously should be removed. I was talking specifically about `loop`, it is not needed.

Reply:
The python-can SocketCanDaemonBus class does not use mtu or bitrate, and the docs do not specify needing them, so I'm going to remove them from my implementation.